File: api/src/models/FinancialYear.js
import mongoose from 'mongoose';
import { getConfigConnection } from '../config/db.js';

const financialYearSchema = new mongoose.Schema(
  {
    financialYearName: { type: String, required: true, trim: true },
    startDate: { type: Date, required: true },
    endDate: { type: Date, required: true },
    databaseName: { type: String, required: true, trim: true },
    isActive: { type: Boolean, default: true },
    closed: { type: Boolean, default: false },
    previousYearId: { type: mongoose.Schema.Types.ObjectId, ref: 'FinancialYear', default: null },
    createdDate: { type: Date, default: () => new Date() },
    createdBy: { type: String, default: null }
  },
  { collection: 'FinancialYear' }
);

financialYearSchema.index({ financialYearName: 1 }, { unique: true });
financialYearSchema.index({ databaseName: 1 }, { unique: true });
financialYearSchema.index({ startDate: 1, endDate: 1 }, { unique: true });

const conn = getConfigConnection();
export const FinancialYear = conn.model('FinancialYear', financialYearSchema);

File: ims-api-net/src/Ims.Domain/Config/FinancialYear.cs
namespace Ims.Domain.Config;

public sealed class FinancialYear
{
    public string Id { get; set; } = Common.ObjectIdGenerator.NewId();

    public string FinancialYearName { get; set; } = string.Empty;

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public string DatabaseName { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public bool Closed { get; set; }

    public string? PreviousYearId { get; set; }

    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public string? CreatedBy { get; set; }
}
